File: src/_Frontend/src/datas/approveHistory.ts
import { SubmissionMethod } from "../types";

export type ApproveHistory = {
    id: string,
    expenseId: string,
    date: Date,
    userId: string,
    type: string,
    amount: number,
    submissionMethod: SubmissionMethod | null,
    receiptId: string | null,
    reviewerId: string | null,
    comment: string | null,
    status: string | null,
    workName: string | null,
    startSection: string | null,
    endSection: string | null,
    categoryId: string | null,
    remarks: string | null,
    expenseTypeId: string | null,
    participationNumber: number | null,
    details: string | null
}

File: src/DbMigration/DbMigration/Tables/ApproveHistory.cs
namespace DbMigration.Tables;

/// <summary>承認履歴テーブル</summary>
public class ApproveHistory
{
    /// <summary>ID</summary>
    public Ulid Id { get; set; }

    /// <summary>経費精算ID</summary>
    public Ulid ExpenseId { get; set; }

    /// <summary>ユーザーID</summary>
    public Ulid UserId { get; set; }

    /// <summary>日付</summary>
    public DateTime Date { get; set; }

    /// <summary>コメント</summary>
    public string? Comment { get; set; }

    /// <summary>ステータス</summary>
    public string Status { get; set; } = string.Empty;
}